cub/block/block_run_length_decode.cuh

```diff
@@ -0,0 +1,435 @@
+                                                      const RelativeRunOffsetT &relative_offset)
+  {
+    decode_buffer.uniques[buffer_offset] = unique_item;
+    decode_buffer.offsets[buffer_offset] = relative_offset;
+  }
+
+  /**
+   * [SPECIALIZATION: NORMAL OR OFFSETS]
+   * Populates the run-length decode buffer with <continuation-of-current-run> items
+   */
+  __device__ __forceinline__ void LoadFromDecodeBuffer(ZippedRunLengthDecode (&scan_items)[DECODED_ITEMS_PER_THREAD],
+                                                       UnzippedRunLengthDecodeBuffer &decode_buffer)
+  {
+    RelativeOffsetT run_offset = linear_tid * DECODED_ITEMS_PER_THREAD;
+#pragma unroll
+    for (uint32_t i = 0; i < DECODED_ITEMS_PER_THREAD; i++)
+    {
+      scan_items[i].unique = decode_buffer.uniques[run_offset];
+      scan_items[i].offset = decode_buffer.offsets[run_offset];
+      run_offset++;
+    }
+  }
+
+  /**
+   * [SPECIALIZATION: NORMAL OR OFFSETS]
+   * Populate registers (SOA) to return the results to the user
+   */
+  template <typename UserRelativeOffsetT>
+  __device__ __forceinline__ void PopulateReturnValues(ZippedRunLengthDecode (&scan_items)[DECODED_ITEMS_PER_THREAD],
+                                                       UniqueItemT (&uniques)[DECODED_ITEMS_PER_THREAD],
+                                                       UserRelativeOffsetT (&relative_offset)[DECODED_ITEMS_PER_THREAD])
+  {
+#pragma unroll
+    for (uint32_t i = 0; i < DECODED_ITEMS_PER_THREAD; i++)
+    {
+      uniques[i]         = scan_items[i].unique;
+      relative_offset[i] = scan_items[i].offset;
+    }
+  }
+
+public:
+  struct TempStorage : Uninitialized<_TempStorage>
+  {};
+
+  //---------------------------------------------------------------------
+  // CONSTRUCTOR
+  //---------------------------------------------------------------------
+  __device__ __forceinline__ BlockRunLengthDecode()
+      : temp_storage(PrivateStorage())
+      , linear_tid(RowMajorTid(BLOCK_DIM_X, BLOCK_DIM_Y, BLOCK_DIM_Z))
+  {}
+
+  __device__ __forceinline__ BlockRunLengthDecode(TempStorage &temp_storage)
+      : temp_storage(temp_storage.Alias())
+      , linear_tid(RowMajorTid(BLOCK_DIM_X, BLOCK_DIM_Y, BLOCK_DIM_Z))
+  {}
+
+  /**
+   * @brief Initializes the run-length decoding instances with the given sequence of runs and returns the total
+   * run-length decoded size. Subsequent calls to <b>RunLengthDecode</b> can be used to retrieve the run-length
+   * decoded data.
+   */
+  template <typename RunLengthT, typename OffsetT>
+  __device__ __forceinline__ void Init(UniqueItemT (&unique_items)[RUNS_PER_THREAD],
+                                       RunLengthT (&run_lengths)[RUNS_PER_THREAD],
+                                       OffsetT &total_decoded_size)
+  {
+    // Compute the offset for the beginning of each run
+    RelativeOffsetT run_offsets[RUNS_PER_THREAD];
+#pragma unroll
+    for (uint32_t i = 0; i < RUNS_PER_THREAD; i++)
```

Comment:
`RUNS_PER_THREAD` is `int`, so we can't compare it with `uint32_t`. Can we change the type of `i` to `int` here and below?

Reply:
Good catch. Fixed 👍